feat: fetch container images from manifests

danielskinstad opened the pull request:
`gen_docker-compose` will now fetch container images from the docker-compose manifests if `--images-dir` isn't specified. You can specify the architecture with `--architecture <arch>` and list the available architectures with `--list-architectures`.

Ticket: MEN-8892

Note that the ticket says to specify the registry, but there's no easy way (AFAICT) to do this on a per-image basis in our generator, and you can have images from different registries in the same manifest. However, you can specify a "full path" in the docker-compose, e.g. `quay.io/skopeo/stable` or `registry.mender.io/mender-server-enterprise/iot-manager`, which will work fine.

vpodzime commented on src/gen_docker-compose:
```diff
@@ -184,8 +212,28 @@ function cleanup() {
-cp -r "$images_dir" "${temp_dir}/images"
+if [ -n "${images_dir}" ]; then
+    cp -r "$images_dir" "${temp_dir}/images"
+fi
```
Why not do this in the `if-elif-elif` above? I think it belongs there.

danielskinstad replied:
Sure, we can add an `else` there

vpodzime reviewed (approved):
Looks good to me otherwise.

danielskinstad committed:
feat: fetch container images from manifests

`gen_docker-compose` will now fetch container images from the
docker-compose manifests if `--images-dir` isn't specified. You can
specify the architecture with `--architecture <arch>` and list the available
architectures with `--list-architectures`.

Ticket: MEN-8892

Signed-off-by: Daniel Skinstad Drabitzius <daniel.drabitzius@northern.tech>
